tests/app/views/test_suppliers.py

```diff
@@ -321,3 +322,337 @@ def test_when_supplier_contact_information_email_invalid(self):
+        response = self.update_request({'suppliers': {
+            'new_key': "value",
+            'name': "New Name"
+        }})
+
+        assert_equal(response.status_code, 400)
+
+
+class TestUpdateContactInformation(BaseApplicationTest):
+    def setup(self):
+        super(TestUpdateContactInformation, self).setup()
+
+        with self.app.app_context():
+            payload = self.load_example_listing("Supplier")
+            self.supplier = payload
+            self.supplier_id = payload['id']
+
+            response = self.client.put(
+                '/suppliers/{}'.format(self.supplier_id),
+                data=json.dumps({'suppliers': self.supplier}),
+                content_type='application/json')
+            supplier = json.loads(response.get_data())['suppliers']
+            self.contact_id = supplier['contactInformation'][0]['id']
+
+    def update_request(self, data):
+        return self.client.post(
+            '/suppliers/123456/contact-information/{}'.format(self.contact_id),
+            data=json.dumps(data),
+            content_type='application/json',
+        )
+
+    def test_empty_update(self):
+        response = self.update_request({'contactInformation': {}})
+        assert_equal(response.status_code, 200)
+
+    def test_simple_field_update(self):
+        response = self.update_request({'contactInformation': {
+            'city': "New City"
+        }})
+        assert_equal(response.status_code, 200)
+
+        with self.app.app_context():
+            contact = ContactInformation.query.filter(
+                ContactInformation.id == self.contact_id
+            ).first()
+
+            assert_equal(contact.city, "New City")
+
+    def test_update_response_matches_payload(self):
+        payload = self.load_example_listing("Supplier")
+        response = self.update_request({'contactInformation': {
+            'city': "New City"
+        }})
+        assert_equal(response.status_code, 200)
+
+        payload = payload['contactInformation'][0]
+        payload.update({'city': 'New City'})
+        contact = json.loads(response.get_data())['contactInformation']
+        contact.pop('id')
+
+        assert_equal(contact, payload)
+
+    def test_update_all_fields(self):
+        response = self.update_request({'contactInformation': {
+            "contactName": "New Value",
+            "phoneNumber": "New Value",
+            "email": "new-value@example.com",
+            "website": "example.com",
+            "address1": "New Value",
+            "address2": "New Value",
+            "city": "New Value",
+            "country": "New Value",
+            "postcode": "New Value",
+        }})
+
+        assert_equal(response.status_code, 200)
+
+        with self.app.app_context():
+            contact = ContactInformation.query.filter(
+                ContactInformation.id == self.contact_id
+            ).first()
+
+        assert_equal(contact.contact_name, "New Value")
```

Comment:
It might be nice for the test to have different values for each of the updated fields - so that we are sure that the right fields are getting the right updated values.